library/common/buffer/utility.cc

```diff
@@ -17,10 +17,13 @@ Buffer::InstancePtr toInternalData(envoy_data data) {
   return buf;
 }
 
-envoy_data toBridgeData(Buffer::Instance& data) {
+absl::optional<envoy_data> toBridgeData(Buffer::Instance& data) {
   envoy_data bridge_data;
   bridge_data.length = data.length();
   bridge_data.bytes = static_cast<uint8_t*>(malloc(sizeof(uint8_t) * bridge_data.length));
```

Comment:
We need to guard against data length 0, for which malloc might still return nullptr (but this would be valid).

Reply:
done here and everywhere else we malloc for the bridge types in common code.